MT-CTF/capturetheflag
# Sniper class

- Spawns with
  - `sniper_rifles:rifle_762`
  - 2x `grenades:smoke`
- Allowed guns
  - Pistol
  - SMG
  - 7.62mm rifle
  - Magnum rifle
- Unlocked crafting recipes
  - 7.62mm rifle
  - Magnum rifle
- Shooter perks
  - +25% range
  - +25% faster shooting
- Normal HP and speed.
- Temporarily uses the now-unused rocketeer skin.

****

This PR contains another commit that makes `give_initial_stuff` split itemstacks if the stack count is greater than the item's `stack_max`. This is very useful when adding more than one of a non-stackable item in `initial_stuff` (like weapons and grenades).
Note: Multiple stuff providers providing the same item is still disallowed, as was originally intended.

PR is ready for review.

### Testing instructions

- Verify that snipers are given the specified initial items, and use the rocketeer skin (the skin will mostly be changed later).
- Also verify that the given smoke grenades aren't stacked in the same slot - they should occupy individual slots each.
- Verify that other classes (esp. shooter) can't use the sniper rifles.
